src/library_gl.js

```diff
@@ -665,46 +665,116 @@ var LibraryGL = {
 
-      gl.bindTexture(gl.TEXTURE_2D, prevTextureBinding);
-      gl.activeTexture(prevActiveTexture);
-      // prevEnableVertexAttribArray?
-      // prevVertexAttribPointer?
-      gl.bindBuffer(gl.ARRAY_BUFFER, prevVB);
-      gl.useProgram(prevProgram);
+          gl.bindVertexArray(prevVAO);
+          gl.deleteVertexArray(vao);
+        } else {
+          var prevVertexAttribPointer = {
+            buffer: gl.getVertexAttrib(context.blitPosLoc, gl.VERTEX_ATTRIB_ARRAY_BUFFER_BINDING),
+            size: gl.getVertexAttrib(context.blitPosLoc, gl.VERTEX_ATTRIB_ARRAY_SIZE),
+            stride: gl.getVertexAttrib(context.blitPosLoc, gl.VERTEX_ATTRIB_ARRAY_STRIDE),
+            type: gl.getVertexAttrib(context.blitPosLoc, gl.VERTEX_ATTRIB_ARRAY_TYPE),
+            normalized: gl.getVertexAttrib(context.blitPosLoc, gl.VERTEX_ATTRIB_ARRAY_NORMALIZED),
+            pointer: gl.getVertexAttribOffset(context.blitPosLoc, gl.VERTEX_ATTRIB_ARRAY_POINTER),
+          };
+          var maxVertexAttribs = gl.getParameter(gl.MAX_VERTEX_ATTRIBS);
+          var prevVertexAttribEnables = [];
+          for (var i = 0; i < maxVertexAttribs; ++i) {
+            var prevEnabled = gl.getVertexAttrib(i, gl.VERTEX_ATTRIB_ARRAY_ENABLED);
+            var wantEnabled = i == context.blitPosLoc;
+            if (prevEnabled && !wantEnabled) {
+              gl.disableVertexAttribArray(i);
+            }
+            if (!prevEnabled && wantEnabled) {
+              gl.enableVertexAttribArray(i);
+            }
+            prevVertexAttribEnables[i] = prevEnabled;
+          }
+
+          draw();
+
+          for (var i = 0; i < maxVertexAttribs; ++i) {
+            var prevEnabled = prevVertexAttribEnables[i];
+            var nowEnabled = i == context.blitPosLoc;
+            if (prevEnabled && !nowEnabled) {
+              gl.enableVertexAttribArray(i);
+            }
+            if (!prevEnabled && nowEnabled) {
+              gl.disableVertexAttribArray(i);
+            }
+          }
+          gl.bindBuffer(gl.ARRAY_BUFFER, prevVertexAttribPointer.buffer);
+          gl.vertexAttribPointer(context.blitPosLoc,
+                                 prevVertexAttribPointer.size,
+                                 prevVertexAttribPointer.type,
+                                 prevVertexAttribPointer.normalized,
+                                 prevVertexAttribPointer.stride,
+                                 prevVertexAttribPointer.offset);
+        }
+
+        if (prevStencilTest) gl.enable(gl.STENCIL_TEST);
+        if (prevDepthTest) gl.enable(gl.DEPTH_TEST);
+        if (prevCullFace) gl.enable(gl.CULL_FACE);
+        if (prevBlend) gl.enable(gl.BLEND);
+
+        gl.bindTexture(gl.TEXTURE_2D, prevTextureBinding);
+        gl.activeTexture(prevActiveTexture);
+        gl.bindBuffer(gl.ARRAY_BUFFER, prevVB);
+        gl.useProgram(prevProgram);
+      }
       gl.bindFramebuffer(gl.FRAMEBUFFER, prevFbo);
+      if (prevScissorTest) gl.enable(gl.SCISSOR_TEST);
```

Comment:
note to self: this should be if(prevScissor)... // or change vars up on lines 668-669 above...

Reply:
oops, thank you for finding this!